Apps/UnitTests/Shared/Shared.cpp

```diff
@@ -11,41 +9,81 @@
 #include <Babylon/ScriptLoader.h>
 #include <chrono>
 #include <thread>
-#include <napi/env.h>
+#include <optional>
+#include <future>
+#include <iostream>
 
-Babylon::Graphics::Configuration deviceTestConfig{};
+namespace
+{
+    Babylon::Graphics::Configuration deviceConfig{};
 
-TEST(JSTest, JavaScriptTests)
+    const char* EnumToString(Babylon::Polyfills::Console::LogLevel logLevel)
+    {
+        switch (logLevel)
+        {
+            case Babylon::Polyfills::Console::LogLevel::Log:
+                return "log";
+            case Babylon::Polyfills::Console::LogLevel::Warn:
+                return "warn";
+            case Babylon::Polyfills::Console::LogLevel::Error:
+                return "error";
+        }
+
+        return "unknown";
+    }
+}
+
+TEST(JavaScript, All)
 {
-    std::promise<int32_t> exitCode;
-    Babylon::Graphics::Device device{deviceTestConfig};
+    // Change this to true to wait for the JavaScript debugger to attach (only applies to V8)
+    constexpr const bool waitForDebugger = false;
+
+    std::promise<int32_t> exitCodePromise;
+
+    Babylon::Graphics::Device device{deviceConfig};
 
     std::optional<Babylon::Polyfills::Canvas> nativeCanvas;
 
-    Babylon::AppRuntime runtime{};
-    runtime.Dispatch([&exitCode, &device, &nativeCanvas](Napi::Env env)
+    Babylon::AppRuntime::Options options{};
+
+    options.UnhandledExceptionHandler = [&exitCodePromise](const Napi::Error& error) {
+        std::cerr << "[Uncaught Error] " << error.Get("stack").As<Napi::String>().Utf8Value() << std::endl;
+        std::cerr.flush();
+
+        exitCodePromise.set_value(-1);
+    };
+
+    if (waitForDebugger)
     {
+        std::cout << "Waiting for debugger..." << std::endl;
+        options.WaitForDebugger = true;
+    }
+
+    Babylon::AppRuntime runtime{options};
+
+    runtime.Dispatch([&exitCodePromise, &device, &nativeCanvas](Napi::Env env) {
         device.AddToJavaScript(env);
 
         Babylon::Polyfills::XMLHttpRequest::Initialize(env);
-        Babylon::Polyfills::Console::Initialize(env, [](const char* message, auto)
-        {
-            printf("%s", message);
-            fflush(stdout);
+        Babylon::Polyfills::Console::Initialize(env, [](const char* message, Babylon::Polyfills::Console::LogLevel logLevel) {
+            std::cout << "[" << EnumToString(logLevel) << "] " << message << std::endl;
+            std::cout.flush();
         });
         Babylon::Polyfills::Window::Initialize(env);
         nativeCanvas.emplace(Babylon::Polyfills::Canvas::Initialize(env));
         Babylon::Plugins::NativeEngine::Initialize(env);
-        
-        env.Global().Set("SetExitCode", Napi::Function::New(env, [&exitCode](const Napi::CallbackInfo& info)
-        {
-            Napi::Env env = info.Env();
-            exitCode.set_value(info[0].As<Napi::Number>().Int32Value());
-        }, "SetExitCode"));
+
+        auto setExitCodeCallback = Napi::Function::New(
```

Comment:
Why add the variable?

Reply:
Formatting. Clang-format indents like crazy otherwise.